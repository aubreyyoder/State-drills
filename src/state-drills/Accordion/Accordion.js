import React from "react";

class Accordion extends React.Component {
  static defaultProps = { sections: [] };
  state = {
    currentSectionIndex: null
  };

  handleButtonClick(index) {
    this.setState({ currentSectionIndex: index });
  }

  renderItem(section, index, currentSectionIndex) {
    return (
      <li className="Accordion_item" key={index}>
        <button type="button" onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        {currentSectionIndex === index && <p>{section.content}</p>}
      </li>
    );
  }

  render() {
    const { currentSectionIndex } = this.state;
    const { sections } = this.props;
    return (
      <div>
        <ul className="Accordion">
          {sections.map((section, index) =>
            this.renderItem(section, index, currentSectionIndex)
          )}
        </ul>
      </div>
    );
  }
}

export default Accordion;
