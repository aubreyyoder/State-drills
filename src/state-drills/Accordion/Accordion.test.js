import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Accordion from "./Accordion";

describe("Accordion Component", () => {
  const sectionsProp = [
    {
      title: "Section 1",
      content:
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque."
    },
    {
      title: "Section 2",
      content:
        "Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit."
    },
    {
      title: "Section 3",
      content:
        "Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur."
    }
  ];

  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders empty li when prop isn't supplied", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders no section as active by default", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("opens a clicked section", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("only opens one sections at a time", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    wrapper
      .find("button")
      .at(2)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
