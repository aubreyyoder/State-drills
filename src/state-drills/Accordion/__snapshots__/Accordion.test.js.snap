// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one sections at a time 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion_item"
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion_item"
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="Accordion_item"
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p>
        Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion_item"
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion_item"
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
      <p>
        Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      className="Accordion_item"
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders empty li when prop isn't supplied 1`] = `
<div>
  <ul
    className="Accordion"
  />
</div>
`;

exports[`Accordion Component renders no section as active by default 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion_item"
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion_item"
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="Accordion_item"
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
